Mathlib/Geometry/Manifold/IntegralCurve/UniformTime.lean

```diff
@@ -0,0 +1,200 @@
+  intros t' ht'
+  by_cases hlt : |t'| + 1 < a
+  · exact eqOn_of_isIntegralCurveOn_Ioo hv γ hγx hγ
+      (by positivity) (le_of_lt hlt) (abs_lt.mp <| lt_add_one _)
+  · exact eqOn_of_isIntegralCurveOn_Ioo hv γ hγx hγ
+      (neg_lt_self_iff.mp <| lt_trans ht'.1 ht'.2) (not_lt.mp hlt) ht' |>.symm
+
+/-- If for every `a : ℝ`, there exists an integral curve defined on `Ioo (-a) a`, all with the
+  same starting point `x`, then there exists a global integral curve starting at `x`. -/
+lemma exists_integralCurve_of_exists_isIntegralCurveOn_Ioo [BoundarylessManifold I M]
+    (hv : ContMDiff I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M))) {x : M}
+    (h : ∀ a, ∃ γ, γ 0 = x ∧ IsIntegralCurveOn γ v (Ioo (-a) a)) :
+    ∃ γ, γ 0 = x ∧ IsIntegralCurve γ v := by
+  refine ⟨fun t' ↦ choose (h (|t'| + 1)) t', (choose_spec (h (|0| + 1))).1, ?_⟩
+  intro t
+  apply HasMFDerivAt.congr_of_eventuallyEq (f := choose (h (|t| + 1)))
+  · apply (choose_spec (h (|t| + 1))).2 t
+    rw [mem_Ioo, ← abs_lt]
+    exact lt_add_one _
+  · rw [Filter.eventuallyEq_iff_exists_mem]
+    refine ⟨Ioo (-(|t| + 1)) (|t| + 1), ?_,
+      exists_integralCurve_of_exists_isIntegralCurveOn_Ioo_eqOn_aux hv h⟩
+    · have := lt_add_of_pos_right |t| zero_lt_one
+      rw [abs_lt] at this
+      exact Ioo_mem_nhds this.1 this.2
+
+lemma exists_isIntegralCurve_iff_exists_isIntegralCurveOn_Ioo [BoundarylessManifold I M]
+    (hv : ContMDiff I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M))) (x : M) :
+    (∃ γ, γ 0 = x ∧ IsIntegralCurve γ v) ↔
+      ∀ a, ∃ γ, γ 0 = x ∧ IsIntegralCurveOn γ v (Ioo (-a) a) :=
+  ⟨fun ⟨γ, h1, h2⟩ _ ↦ ⟨γ, h1, h2.isIntegralCurveOn _⟩,
+   exists_integralCurve_of_exists_isIntegralCurveOn_Ioo hv⟩
+
+lemma piecewise_eqOn_symm [BoundarylessManifold I M]
+    (hv : ContMDiff I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M)))
+    {a b a' b' : ℝ} (hγ : IsIntegralCurveOn γ v (Ioo a b))
+    (hγ' : IsIntegralCurveOn γ' v (Ioo a' b'))
+    (ht₀ : t₀ ∈ Ioo a b ∩ Ioo a' b') (h : γ t₀ = γ' t₀) :
+    EqOn (piecewise (Ioo a b) γ γ') γ' (Ioo a' b') := by
+  intros t ht
+  suffices H : EqOn γ γ' (Ioo (max a a') (min b b')) by
+    by_cases hmem : t ∈ Ioo a b
+    · rw [piecewise, if_pos hmem]
+      apply H
+      simp [ht.1, ht.2, hmem.1, hmem.2]
+    · rw [piecewise, if_neg hmem]
+  apply isIntegralCurveOn_Ioo_eqOn_of_contMDiff_boundaryless _ hv
+    (hγ.mono (Ioo_subset_Ioo (le_max_left ..) (min_le_left ..)))
+    (hγ'.mono (Ioo_subset_Ioo (le_max_right ..) (min_le_right ..))) h
+  exact ⟨max_lt ht₀.1.1 ht₀.2.1, lt_min ht₀.1.2 ht₀.2.2⟩
+
+/-- The extension of an integral curve by another integral curve is an integral curve.
+
+  If two integral curves are defined on overlapping open intervals, and they agree at a point in
+  their common domain, then they can be patched together to form a longer integral curve.
+
+  This is stated for manifolds without boundary for simplicity. We actually only need to assume that
```

Comment:
Can you record a TODO for this, e.g. in the module doc-string? You can also mention compactness there...